### ISSUE TITLE
Using local terraform.tfstate with new flags

### ISSUE THREAD

**Issue by rirze** [infracost/infracost#1487]
Hi,

I'm currently testing `infracost` for a private repo. I can currently test it with a downloaded `terraform.tfstate` file, but I'm unable to get it to work with my current understanding of the flags. 

I've tried:
```
infracost breakdown --terraform-use-state --path /path/to/tf/dir
```
That results in:
```
 ✖ Running terraform show 

Error: terraform show failed: exit status 1
<some error about accessing state over http>
```
Which I expect, since the location of terraform state over HTTP is inaccessible from my testing location. Am I misunderstanding what is required for `infracost` to work or is there a combination of flags that allow me to generate a report solely from a locally downloaded terraform state file?

**Comment by vdmgolub** [comment]
Hi @rirze! Interesting, this flag just makes Infracost CLI run `terraform show` instead of `terraform plan`. Seems like Terraform can't find the state file.

One way to unblock you is generating Terraform's plan JSON and feed it to Infracost:
1. Run `terraform show -json my-state.tfstate > plan.json`
2. Use it with Infracost `infracost breakdown --path plan.json`

For reference, the docs: https://www.infracost.io/docs/features/cli_commands/#option-2-terraform-plan-json.

Could you try that and see if it works? :)

**Comment by rirze** [comment]
Gotcha, your reply makes it seems like I need to run either `terraform show` or `terraform plan` if I plan on testing this locally. 

I don't think that will work for this demo then, since `terraform show` seems to still require access to the state (which is accessed in a private domain over HTTP). 

I trying to evaluate `infracost` without devoting resources to self-hosting it, seems like it will require some more tuning from my part to get a report. Maybe I can edit the downloaded state file to just reference itself as local state....?

**Comment by vdmgolub** [comment]
@rirze There is a way without using Terraform! We've recently released a feature to parse HCL files. It won't give 100% accuracy (yet), but it should provide a good idea how Infracost works:

```
infracost breakdown --path /path/to/tf/dir --terraform-parse-hcl
```

I need to warn you though that this is an experimental feature and there may be discrepancies between TF plan and the HCL parser. We're working on fixing those, but hopefully this can unblock you right now.

**Comment by rirze** [comment]
... Thank you so much for offering so many ways to help-- I hate to say this, but we use the CDK for Terraform to generate JSON Terraform files so there is no HCL in this case. Just JSON.  

Unless I can find a JSON->HCL converter (which I can't imagine why anyone would build).

**Comment by rirze** [comment]
Well, I found `yj`, which allows for JSON->HCLv1 conversion (with many other formats as well) and I was able to create a HCL file which I then fed into `infracost`. 

I obtained a reasonable report, so I'm pretty satisfied with this approach for now. Thanks for the help.

**Comment by vdmgolub** [comment]
@rirze Oh, I'm sorry, I think I missed your comment from yesterday in another issue. Have you tried what Ali suggested [here](https://github.com/infracost/infracost/issues/1219#issuecomment-1075563445)? You don't need to use `--terraform-use-state` flag, just point `--path` to the directory with the JSON files.

**Comment by rirze** [comment]
@vdmgolub right, I can't run that command successfully since that command will need to be run internally in our environment, and not externally with this demo I'm doing. 

I will update that ticket once I can create a dummy project with the CDK and run `infracost` as instructed.

**Comment by vdmgolub** [comment]
Got it! Thank you for the explanation! Hopefully more people bring their ideas in #1219 and we can make it work in future. :) Good luck with the demo and thank you for trying out Infracost! :)
